### PR TITLE
Update Program Hashes for Ops using Mem config

### PR DESCRIPTION
**Pull request by ntarafdar**
solves: https://github.com/tenstorrent-metal/tt-metal/issues/4950

### CODE REVIEW

File: tt_eager/tt_dnn/op_library/transpose/wh_multi_core/transpose_wh_op_multi_core.cpp [hunk 1]
```diff
@@ -362,16 +362,16 @@ operation::ProgramWithCallbacks transpose_wh_multi_core_sharded(const Tensor &a,
         const std::vector<Tensor>& output_tensors
     ) {
 
-        const auto& src_tensor = input_tensors.at(0);
-        const auto& dst_tensor = output_tensors.at(0);
+        const auto src_tensor = input_tensors.at(0);
+        const auto dst_tensor = output_tensors.at(0);
```
Comment (tt-aho) [inline tt_eager/tt_dnn/op_library/transpose/wh_multi_core/transpose_wh_op_multi_core.cpp:366]: Why do we remove ref here?

Comment (tarafdarTT) [reply tt_eager/tt_dnn/op_library/transpose/wh_multi_core/transpose_wh_op_multi_core.cpp:366]: I think it was unnecessary here tbh, but below where we get the shard_spec from ref and use it in bounding boxes, I think the variable it was referring to goes out of bounds and I end up getting a set fault at the `all_cores.bounding_box` line.

Comment (tarafdarTT) [reply tt_eager/tt_dnn/op_library/transpose/wh_multi_core/transpose_wh_op_multi_core.cpp:366]: I can set the ref back for this and just leave it for shard_spec

**Review by tt-aho — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]
LGTM

**Review by arakhmati — APPROVED** [review]

**Pull request merged**